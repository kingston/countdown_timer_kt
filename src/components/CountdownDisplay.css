.countdown-display {
  font-size: 100px;
  font-weight: bold;
  line-height: 100px;
}
