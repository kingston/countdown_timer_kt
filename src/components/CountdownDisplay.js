import React from 'react';
import PropTypes from 'prop-types';

import './CountdownDisplay.css';

function CountdownDisplay({ secondsRemaining, totalSeconds }) {
  const minutes = Math.floor(secondsRemaining / 60)
    .toString()
    .padStart(2, '0');
  const seconds = (secondsRemaining % 60).toString().padStart(2, '0');
  return (
    <div className="countdown-display">
      {minutes}:{seconds}
    </div>
  );
}

CountdownDisplay.propTypes = {
  secondsRemaining: PropTypes.number.isRequired,
  totalSeconds: PropTypes.number.isRequired,
};

export default CountdownDisplay;
