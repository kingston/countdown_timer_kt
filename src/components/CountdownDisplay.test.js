import React from 'react';
import CountdownDisplay from './CountdownDisplay';
import { shallow } from 'enzyme';

it('renders initial state', () => {
  const initialState = shallow(
    <CountdownDisplay secondsRemaining={0} totalSeconds={0} />
  );
  expect(initialState).toMatchSnapshot();
});
