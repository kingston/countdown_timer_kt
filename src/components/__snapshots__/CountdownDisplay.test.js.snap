// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders initial state 1`] = `
<div
  className="countdown-display"
>
  00
  :
  00
</div>
`;
